Upgrade to be a Polymer 1.9-2.0 hybrid component

Mihaiii opened the pull request:
Upgrade to be a Polymer 1.9-2.0 hybrid component.
For Issue #28 and https://github.com/StarcounterApps/KitchenSink/pull/155 .

Commit:
Upgrade to be a Polymer 1.9-2.0 hybrid component; for Issue #28 and Starcounter/RebelsLounge#140

warpech commented on bower.json:
```diff
@@ -24,10 +24,10 @@
   ],
   "dependencies": {
-    "polymer": "polymer/polymer#^1.1.0"
+    "polymer": "polymer/polymer#~1.9"
```
Polymer recommends such selector for hybrid components: `"polymer": "Polymer/polymer#1.9 - 2"`

See: https://www.polymer-project.org/2.0/docs/devguide/hybrid-elements

warpech reviewed (commented):
Were you able to run the demo page both with Polymer 1.9 and 2.0?

Mihaiii commented:
I don't know how urgent this is, but after spending some time on it, I decided to focus on other tasks that I have at the moment and then come back to it. 

The changes that are required aren't only the type of changes that were used when the same upgrade was done for `juicy-select`.

Please let me know if this should have high priority and I will focus again on it.

warpech commented:
In general, it is important to finish started PRs, because otherwise unmerged code is left at a branch at starts rotting.

In case of this library it is important to upgrade to hybrid Polymer because it is used in two of our prefab apps (KitchenSink and Products) and we want to upgrade these apps to Polymer 2.0 few weeks from now.

Mihaiii commented:
I see, thank you!

I spent some time trying to figure out what I should do here and couldn't get to any acceptable decision.

After I made similar changes to the ones in `juicy-select`, I put 2 break points:
- one [here ](https://github.com/Juicy/x-breadcrumb/blob/master/usage.html#L28)
and
- one [here](https://github.com/Juicy/x-breadcrumb/blob/master/pim-x-element.html#L207)

With 1.9.2 the first one hits first, but with 2.0.1 the 2nd one hits first.
The order of execution changes just by changing the version of Polymer, which is strange and I could use some help with this if it's urgent.

I mean, I can fix it, but it won't be "nice code" :) .

warpech commented:
Please come up with "not nice code" then and we can review in a PR :)

Commit:
This file causes more harm than good. Use polyserve for local testing

For Issue #30

Commit:
Standard updates for hybrid element.

This won't work is just for coming back in case further commits won't be accepted.

Commit:
Update readme.md and bower.json

Commit:
An attempt to solve the problem that was described in a comment of this PR

Commit:
Remove since it messes CSS and works without

Mihaiii commented:
This is ready for a review.

Commit:
Undo changes in latest commit and update styles instead

Commit:
Fix README.md

warpech commented on bower.json:
```diff
@@ -24,10 +24,10 @@
-    "polymer": "^1.1.0",
+    "polymer": "~1.9",
     "webcomponentsjs": "^0.7.18"
   }
```
Polymer gives me warnings that the above is an "unneccessary resolution" and installs Polymer 2 anyway:

![2017-07-25 10_36_51-command prompt](https://user-images.githubusercontent.com/566463/28563131-3d8821d0-7125-11e7-8eec-0c0c8ad32c35.png)

Mihaiii replied:
Aren't bower's resolutins taken into consideration only in case of conflict? I'm genuinely asking. https://jaketrent.com/post/bower-resolutions/#bower-json-resolutions

warpech replied:
I think so. But why keep `resolutions` in `bower.json` if there is no conflict?

warpech committed:
simplification proposal

warpech commented on pim-x-element.html:
```diff
@@ -194,8 +205,20 @@
+                            this.addData();
+                        });
+                    }
+                },
```
Does this only exist to support `console.log`? I might be wrong, but it looks like this can be simplified. I made a commit on a separate branch, please review: https://github.com/Juicy/x-breadcrumb/commit/909f318b351af4cd4ff5126376eacecbdcb789a5

warpech reviewed (approved):
Looking good, just two comments. I only tested with Polymer 2. Did you test with KitchenSink (Polymer 1)?

Mihaiii commented:
I reviewed the changes you made, they work great.
I tested in Kitchensink and everything is ok.

Let me know if I can merge since the topic of bower's resolutions is still open (I left a reply to your comment).

warpech commented:
Resolutions should be saved in `bower.json` only if there is some conflict. If `x-breadcrumb` doesn't have any internal conflict, but still there is some conflict in KitchenSink, then `resolutions` should be added to KitchenSink's `bower.json`.

So for me it looks like `resolutions` can be removed, `update/polymer1.9-simplification` merged to `update/polymer1.9` and `update/polymer1.9` to `develop`.

Commit:
Remove as unnecessary

Mihaiii commented:
I merged that commit into this branch yesterday.
Merging this branch into `master` as there isn't a `develop` branch.
